tests/workflow/test_construct_batch.py

```diff
@@ -320,6 +320,7 @@ def test_device_transforms(self, level):
         assert len(batch) == 1
         assert fn(("a",)) == ("a",)
 
+    @pytest.mark.usefixtures("use_legacy_opmath")
```

Comment:
So we continue to test what we are needing to test, lets switch returning the `expval(H)` to

```
return qml.expval(qml.PauliX(0)), qml.expval(qml.PauliY(0))
```

instead, as `split_non_commuting` should still be applied.

Reply:
I get this error with that 😢 
```
@qml.transforms.merge_rotations
@qml.qnode(qml.device("default.qubit.legacy", wires=2, shots=50))
def circuit(order):
    qml.Permute(order, wires=(0, 1, 2))
    qml.RX(0.5, wires=0)
    qml.RX(-0.5, wires=0)
    return [qml.expval(qml.PauliX(0)), qml.expval(qml.PauliY(0))]

batch, fn = qml.workflow.construct_batch(circuit, level=None)((2, 1, 0))
```

```
QuantumFunctionError: Only observables that are qubit-wise commuting Pauli words can be returned on the same wire, some of the following measurements do not commute:
[expval(X(0)), expval(Y(0))]
```

Reply:
offf... that's bad.  Mind opening a bug report? 

I feel like this is a duplicate of several other bugs we've seen  This seems to be a duplicate of #5316 fundamentally, but it's probably worth tracking that this fundamental issue is causing multiple issues throughout pennylane.

In the meantime, we can switch the test to:
```
@qml.transforms.cancel_inverses
@qml.qnode(qml.device("default.qubit.legacy", wires=2, shots=50))
def circuit(order):
    qml.Permute(order, wires=(0, 1, 2))
    qml.X(0)
    qml.X(0)
    return [qml.expval(qml.PauliX(0)), qml.expval(qml.PauliY(0))]

batch, fn = qml.workflow.construct_batch(circuit, level=None)((2, 1, 0))
```